functors_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,29 @@
 #include "impl.h"
 
+BeerBrand BeerOrganizer::operator()() {
+    if (static_cast<size_t>(BeerBrand::Max) == currentIndex)
+        currentIndex = 1;
+
+    return static_cast<BeerBrand>(currentIndex++);
+}
+
+bool isExpensiveBeer(const BeerBrand& brand)
+{
+    std::vector<BeerBrand> expensive{BeerBrand::HoeGaarden, BeerBrand::Corona};
+
+    return std::find(expensive.begin(), expensive.end(), brand) != std::end(expensive);
+}
+
+Cocktail MixingPolicy::mix(AlcoholDrink alcohol, NonAlcoholDrink nonAlcohol)
+{
+    using Recipe = std::tuple<AlcoholDrink, NonAlcoholDrink>;
+    using Ingredients = std::pair<AlcoholDrink, NonAlcoholDrink>;
+
+    std::map<Recipe, Cocktail> cookbook{{Ingredients(AlcoholDrink::Gin, NonAlcoholDrink::LimeJuice), Cocktail::Gimlet},
```

Comment:
1. In such cases, unordered_map should be used, as you don't need the order
2. Why you need to recreate this object every time function is called?
3. Mark it as const

Reply:
Changed to static const unordered_map.